doc/modules/calibration.rst

```diff
@@ -111,30 +123,68 @@ predicted probabilities of the `k` estimators in the
-non-parametric isotonic regressor, which outputs a step-wise non-decreasing
-function (see :mod:`sklearn.isotonic`).
-
-An already fitted classifier can be calibrated by setting `cv="prefit"`. In
-this case, the data is only used to fit the regressor. It is up to the user
+Alternatively an already fitted classifier can be calibrated by setting
+`cv="prefit"`. In this case, the data is not split and all of it is used to
+fit the regressor. It is up to the user
 make sure that the data used for fitting the classifier is disjoint from the
 data used for fitting the regressor.
 
-:class:`CalibratedClassifierCV` can calibrate probabilities in a multiclass
-setting if the base estimator supports multiclass predictions. The classifier
-is calibrated first for each class separately in a one-vs-rest fashion [4]_.
-When predicting probabilities, the calibrated probabilities for each class
+:class:`CalibratedClassifierCV` supports the use of two 'calibration'
+regressors: 'sigmoid' and 'isotonic'. Both these regressors only
+support 1-dimensional data (e.g., binary classification output) but are
+extended for multiclass classification if the `base_estimator` supports
+multiclass predictions. For multiclass predictions,
+:class:`CalibratedClassifierCV` calibrates for
+each class separately in a :ref:`ovr_classification` fashion [4]_. When
+predicting
+probabilities, the calibrated probabilities for each class
 are predicted separately. As those probabilities do not necessarily sum to
 one, a postprocessing is performed to normalize them.
 
 The :func:`sklearn.metrics.brier_score_loss` may be used to evaluate how
 well a classifier is calibrated.
 
+Sigmoid
+^^^^^^^
+
+The sigmoid regressor is based on Platt's logistic model [3]_:
+
+.. math::
+       p(y_i = 1 | f_i) = \frac{1}{1 + \exp(A f_i + B)}
+
+where :math:`y_i` is the true label of sample :math:`i` and :math:`f_i`
+is the output of the classifier for sample :math:`i`. :math:`A` and :math:`B`
+are real numbers to be determined when fitting the regressor via maximum
+likelihood.
+
+The sigmoid method is biased in that it assumes the :ref:`calibration curve
+<calibration_curve>` of the un-calibrated model has a sigmoid shape and is
+symmetrical [1]_. It is thus most effective when the un-calibrated model is
+over-confident and has similar over-confidence errors for both high and low
+output errors. The symmetry assumption is of concern in highly imbalanced
+classification as un-calibrated classifiers can have asymmetric calibration
+errors.
+
+Isotonic
+^^^^^^^^
+
+The 'isotonic' method fits a non-parametric isotonic regressor, which outputs
+a step-wise non-decreasing function (see :mod:`sklearn.isotonic`). It
+minimizes:
+
+.. math::
+       \sum_{i=1}^{n} (y_i - f_i)^2 : f_i \leq f_{i+1}\quad \forall i \{1,..., n-1\}
+
+where :math:`y_i` is the true label of sample :math:`i` and :math:`f_i`
+is the output of the classifier for sample :math:`i`. This method is more
```

Comment:
```suggestion
is the calibrated output of the classifier for sample :math:`i`. This method is more
```

Reply:
...are you sure? I am confused now

Reply:
this is the function that is minimized to find the isotonic function, so should be the output of the classifier..?

Reply:
We're already using `f_i` above to define the output of the un-calibrated classifier.

The formula should be

 \sum_{i=1}^{n} (y_i - \hat{f}_i)^2

where \hat{f} is as @ogrisel suggested (the calibrated probability)

And the constraint is that \hat{f}_i >= \hat{f}_j whenever f_i >= f_j

Reply:
Indeed sorry for the confusion.